File: codigos/chaves.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {
    int n;
    cin >> n;
    cin.ignore();

    stack<char> pilha;

    for (int i = 0; i < n; i++) {
        string palavra;
        getline(cin, palavra);

        for (char j : palavra) {
            if (j == '{') {
                pilha.push('{');
            }
            else if (j == '}') {
                if (pilha.empty()) {
                    cout << "N" << endl;
                    return 0;
                }

                pilha.pop();
            }
        }
    }

    if (pilha.empty()) {
        cout << "S" << endl;
    } else {
        cout << "N" << endl;
    }

    return 0;
}

File: codigos/Python/chaves.py
def main():
    n = int(input())
    pilha = []

    for _ in range (n):
        palavra = input()

        for letra in palavra:
            if letra == '{':
                pilha.append('{')
            elif letra == '}':
                if not pilha:
                    print("N")
                    return

                pilha.pop()

    if not pilha:
        print("S")
    else:
        print("N")


if __name__ == "__main__":
    main()
